### PR TITLE
feat: updated composer and docker file to PHP 8.1 and Nov LTS 2022

### PR DESCRIPTION
**Pull request by mike-ionut-mihai-sandu-tao**
## Release-2021-12

- Merge the Rich Passage Authoring functionality
- Add RTL languages on item authoring
- A priori Cut Score Exposure
- Readability Tool - Tool Activation
- Highlighted text on passages to persist across items
- Selection tool - select & move
- Selection tool - Resize and drag a rectangle or an ellipsis
- Submit AGS on test complete
- Submit AGS on a test launch

## Release-2022-01

- Assets are automatically published to a CDN
- Test Taker Tools
> - Highlighter
> - RTL (Right To Left )

## Release-2022-01

- Improvement on the List Management 
> - Applied a (configurable) limit of 1000 for the maximum number of elements in a list, in order to avoid performance bottlenecks when managing the lists
> - In order to increase the usability and reduce the performance impact when managing big lists, we added pagination (with 20 elements per page) for all lists
> - In order to increase usability and decrease the possibility of creating huge lists by mistake, from now on new lists are created with a single element, while new elements can be added one by one

- Performance improvement for the Advanced Search index processing (only EE customers)
- Improved the event logs whenever a resource is deleted, by adding information about the URI of the deleted resource and the class name where it was deleted from

- Minor Fixes 

> - Fixed an error on delivery execution if one of the items includes an asset with a "&" character within the file path
> - Fixed a situation when the author might be able to save a test with an empty section title. This would lead to errors down the line, when trying to author, preview, export, or publish that test
> - Fixed a general permission problem which lead to some users gaining access to list management even without having the required rights for this.

## Release-2022-03

- Adjust language labels and apply alphabetical ordering on languages forms
- Added support for section shuffling
- Added the ability to select individual cells
- Update the Order interaction mixed announcement
- Better support for LTI 1.3 custom claims
- Improved support for Text-to-Speech while navigating the item with the keyboard
- Add translations for new Norwegian languages

## Release-2022-04

- Math entry - Add gap, nothing happening

## Release-2022-05

- Reduced Media player adjusts to the size of Media Interaction container after losing focus from interaction
- Enable wrap text to images by default in TAO
- Multiple table cell selection

## Release-2022-06

- Graphic gap match - change default matchMax of hotspot and gap image
- PDF media asset is displayed as a link to download, not as a document
- Added the ability to send scores via AGS back to the TAO Backoffice

## Release-2022-07

- Copy/duplicate folders in Authoring
- Added the ability to restart a delivery for a given test-taker, by using the custom parameter "deliverySetting.reset: true"
- Added the ability for a test-taker to insert an image when using the Extended Text interaction (in rich-text mode)
- Added a counter for the number of typed words in the Extended Text interaction, so that both the test-taker and the reviewer can easily track it.
- Added the possibility for Android devices to view PDF files embedded into interactions.

## Release-2022-08

- Added the capability of including image captions to any image while authoring items and passages
- Fixed an issue with File Upload interactions which lead to the size of audio or video media not using the full width available.

## Release-2022-09

- Added the capability of adding horizontal rulers while authoring items and passages.
- Added new REST APIs allowing updating a delivery execution with suspicion level coming from Proctorio, when using this remote proctoring integration.
- Added a new REST API endpoint to retrieve delivery execution information.

## Release-2022-10

- Fixed a rendering issue while authoring passages with images and applying a different color as part of the Text Block Style options.
- Greatly improved the metadata management capability by allowing metadata to be imported together with an item or a test, as long as the property exists in the current location's metadata schema.
- Improved accessibility compliance with WCAG 2.1 AA, by making it easier for people with some disabilities to reach the main content.
- Fixed an issue with extended text interactions which failed to be rendered if using the maxWords constraint.

## Release-2022-11

- Fixed a rendering issue while authoring passages with images and applying a different color as part of the Text Block Style options.
- Implemented full compatibility with PHP 8.1 for all TAO modules.
- Fixed an issue that prevented the results to be sent back to a 3rd party platform (including to TAO) via AGS after a delivery execution is submitted.
- Improved readability for items with large inline choices and multiple columns, by wrapping their content so that it doesn't overlap with the neighboring column.

### CODE REVIEW

**Comment by mike-ionut-mihai-sandu-tao** [comment]
proof on installation 
image

**Comment by mike-ionut-mihai-sandu-tao** [comment]
running containers 
image

**Comment by mike-ionut-mihai-sandu-tao** [comment]
php version 
image

**Comment by mike-ionut-mihai-sandu-tao** [comment]
taoUpdate script run
image

File: docker/phpfpm/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM php:7.4-fpm
+FROM php:8.1.0-fpm
```
Comment (wazelin) [inline docker/phpfpm/Dockerfile:1]: Why not the latest fix, though?
```suggestion
FROM php:8.1.13-fpm
```

Comment (mike-ionut-mihai-sandu-tao) [reply docker/phpfpm/Dockerfile:1]: just because my IDE did not find it as stable , let me take a look and commit that

Comment (mike-ionut-mihai-sandu-tao) [reply docker/phpfpm/Dockerfile:1]: found this : 
https://hub.docker.com/layers/library/php/8.1.13-fpm/images/sha256-febc9b5002105e4ff64de25d30d180f12eca2dc438b0dc1559d8561dcfda1fbc?context=explore
so it looks ok by me

**Review by wazelin — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable) ⚠️ I really can't check that the installation and TAO 3.x itself still work there, but judging by the screenshots attached they do just fine
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: composer.json [hunk 2]
```diff
@@ -17,13 +17,13 @@
...
-    "release 3.5",
+    "release 4.0",
     "computer-based-assessment"
   ],
   "homepage": "http://www.taotesting.com",
   "minimum-stability": "stable",
   "prefer-stable": true,
   "require": {
-    "oat-sa/tao-community": "2021.11.8"
+    "oat-sa/tao-community": "2022.11.6"
```
Comment (gabrielfs7) [inline composer.json:27]: Shall we use this one? https://github.com/oat-sa/tao-community/tree/2022.11.7

**Review by gabrielfs7 — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: composer.json (hunk 2 above)
Comment (wazelin) [reply composer.json:27]: Or maybe even https://github.com/oat-sa/tao-community/tree/v2022.11.9.

It has that latest fix by Focaloid, which looked important.

**Pull request merged**